xclim/indices/fire/_ffdi.py

```diff
@@ -0,0 +1,393 @@
+    else:
+        kbdi0 = xr.full_like(pr.isel(time=0), 0)
+
+    kbdi = xr.apply_ufunc(
+        _keetch_byram_drought_index_pass,
+        pr,
+        tasmax,
+        pr_annual,
+        kbdi0,
+        input_core_dims=[["time"], ["time"], [], []],
+        output_core_dims=[["time"]],
+        dask="parallelized",
+        output_dtypes=[pr.dtype],
+    )
+    kbdi.attrs["units"] = "mm/day"
+    return kbdi
+
+
+def _griffiths_drought_factor_pass(pr, smd, lim):
+    """Pass inputs on to guvectorized function `_griffiths_drought_factor`. DO NOT CALL DIRECTLY, use `griffiths_drought_factor` instead."""
+    # This function is actually only required as xr.apply_ufunc will not receive
+    # a guvectorized function which has the output(s) in its function signature
+    return _griffiths_drought_factor(pr, smd, lim)
+
+
+@declare_units(
+    pr="[precipitation]",
+    smd="[precipitation]",
+)
+def griffiths_drought_factor(
+    pr: xr.DataArray,
+    smd: xr.DataArray,
+    limiting_func: str = "xlim",
+) -> xr.DataArray:
+    """Calculate the Griffiths drought factor based on the soil moisture deficit.
+
+    This method implements the methodology and formula described in :cite:t:`ffdi-finkele_2006`
+    (section 2.2) for calculating the Griffiths drought factor.
+
+    Parameters
+    ----------
+    pr : xr.DataArray
+      Total rainfall over previous 24 hours [mm/day].
+    smd : xarray DataArray
+      Daily soil moisture deficit (often KBDI) [mm/day].
+    limiting_func : {"xlim", "discrete"}
+      How to limit the values of the drought factor.
+      If "xlim" (default), use equation (14) in :cite:t:`ffdi-finkele_2006`.
+      If "discrete", use equation Eq (13) in :cite:t:`ffdi-finkele_2006`.
+
+    Returns
+    -------
+    df : xr.DataArray
+      The limited Griffiths drought factor.
+
+    Notes
+    -----
+    Calculation of the Griffiths drought factor depends on the rainfall over the previous 20 days.
+    Thus, the first available time point in the drought factor returned by this function
+    corresponds to the 20th day of the input data.
+
+    References
+    ----------
+    :cite:cts:`ffdi-griffiths_1999,ffdi-finkele_2006,ffdi-holgate_2017`
+    """
+    pr = convert_units_to(pr, "mm/day")
+    smd = convert_units_to(smd, "mm/day")
+
+    if limiting_func == "xlim":
+        lim = 0
+    elif limiting_func == "discrete":
+        lim = 1
+    else:
+        raise ValueError(f"{limiting_func} is not a valid input for `limiting_func`")
+
+    df = xr.apply_ufunc(
+        _griffiths_drought_factor_pass,
+        pr,
+        smd,
+        kwargs=dict(lim=lim),
+        input_core_dims=[["time"], ["time"]],
+        output_core_dims=[["time"]],
+        dask="parallelized",
+        output_dtypes=[pr.dtype],
+    )
+    df.attrs["units"] = ""
+
+    # First non-zero entry is at the 19th time point since df is calculated
+    # from a 20-day rolling window
+    return df.isel(time=slice(19, None))
```

Comment:
Would returning NaN for the 19 first days make sense? I would prefer that because xclim doesn't usually _crop_ the outputs. One could normally expect to receive the same data structure as the input for indicator that do not perform resampling.